api_modules/sessionize.py

```diff
@@ -1,5 +1,122 @@
 '''
-    Put code for Sessionize API acraping here. 
-    Update this docstring with details of how to use the module.
+    Sessionize API acraping here. 
+
+    created 28.01.2024 by bogdansharpy@gmail.com
+
+    Needs bs4 to be installed: pip install bs4
+
+    Uses publicly accessible Sessionize Embedding API : https://sessionize.com/playbook/embedding
+    API Endpoint:
+        f"https://sessionize.com/api/v2/{event_id}/view/{type}?under=True"
+        type in {"GridSmart", "SpeakerWall", "Speakers", "Sessions"}
+
+    Example of how to use script:
+        python sessionize.py --id amtv2kwb --name "PyCon 2022"
+
+    With no arguments by default will get this events: PyCon 2022, PyCon Limerick 2023, PyCon 2023
+
 '''
 
+import argparse
+import requests
+import csv
+from bs4 import BeautifulSoup
+
+class Sessionize:
+    def __init__(self, events=None) -> None:
+        default_events = { \
+            'amtv2kwb': 'PyCon 2022', \
+            'jb4vxosa': 'PyCon Limerick 2023', \
+            'jbshwhme': 'PyCon 2023', \
```

Comment:
How do you find out the ID of a given event? Also is it possible to get the event name from Sessionize once we know the ID?

Reply:
I take them from similar Sessionize API requests at https://python.ie. I have read the documentation and did not find a way to retrieve the event name by ID. Alongside the Sessionize Embedding API, there exists the regular Sessionize API. However, the available data is the same: schedule, speakers, sessions. Fields specific to the event itself are not available.